auth/couchdb_session_authenticator.go

```diff
@@ -155,6 +160,11 @@ func (a *CouchDbSessionAuthenticator) Authenticate(request *http.Request) error
 // getCookie returns an AuthSession cookie to be used in a request.
 // A new cookie will be fetched from the session end-point when needed.
 func (a *CouchDbSessionAuthenticator) getCookie() (*http.Cookie, error) {
+	a.mu.Lock()
+	defer a.mu.Unlock()
+
+	a.flushRefreshChannel()
+
 	if a.session == nil || !a.session.isValid() {
 		err := a.syncRequestSession()
```

Comment:
This is interesting. I thought about removing `syncRequestSession()`, but then opted to leave it because to me this looked more descriptive on what's going on here. In general I don't like excessive commenting in the code, I find it makes it hard to read, but sure, I can refactor this.

`syncRequestSession` needs to read from the `refresh` channel to make `requestSession()` reusable for both sync and async uses, but sure, I can address this too.

Reply:
done in https://github.com/IBM/cloudant-go-sdk/pull/74/commits/a658b5bff7150a1a8d37d425a673d7584fef3f4c